fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -316,11 +318,80 @@ private void pruneTables(OptExpression tree, TaskContext rootTaskContext, Column
             ruleRewriteIterative(tree, rootTaskContext, new MergeTwoProjectRule());
             rootTaskContext.setRequiredColumns(requiredColumns.clone());
             ruleRewriteOnlyOnce(tree, rootTaskContext, RuleSetType.PRUNE_COLUMNS);
-            context.setEnableLeftRightJoinEquivalenceDerive(true);
             ruleRewriteIterative(tree, rootTaskContext, RuleSetType.PUSH_DOWN_PREDICATE);
         }
     }
 
+    private void ruleBasedMaterializedViewRewritePhase1(OptExpression tree,
+                                                        TaskContext rootTaskContext,
+                                                        ColumnRefSet requiredColumns) {
+        if (!mvRewriteStrategy.enableEagerRewrite) {
+            return;
+        }
+
+        OptimizerTraceUtil.logOptExpression("before RuleBasedMaterializedViewRewrite:\n%s", tree);
```

Comment:
can we duplicate a raw query tree, and do mv rewrite on the copy, and apply the rules we want on it, just like view based mv rewrite